deafrica_waterbodies/waterbodies/polygons/attributes.py

```diff
@@ -0,0 +1,148 @@
+import geohash as gh
+
+
+def assign_unique_ids(polygons):
+    """
+    Function to assign a unique ID to each waterbody polygon.
+
+    Parameters
+    ----------
+    polygons : geopandas.geodataframe.GeoDataFrame
+        GeoDataFrame containing the waterbody polygons.
+
+    Returns
+    -------
+    geopandas.geodataframe.GeoDataFrame
+        GeoDataFrame containing the waterbody polygons with additional columns
+        "UID" and "WB_ID".
+        The "UID" column contains a unique identifier
+        for each polygon.
+        The "WB_ID" column contains an arbitrary numerical ID for each
+        polygon with polygons close to each other numbered similarly.
+    """
+
+    crs = polygons.crs
+
+    # Generate a unique id for each polygon.
+    polygons_with_unique_ids = polygons.to_crs(epsg=4326)
+    polygons_with_unique_ids['UID'] = polygons_with_unique_ids.apply(lambda x: gh.encode(x.geometry.centroid.y, x.geometry.centroid.x, precision=9), axis=1)
+
+    # Check that our unique ID is in fact unique
+    assert polygons_with_unique_ids['UID'].is_unique
+
+    # Make an arbitrary numerical ID for each polygon. We will first sort the dataframe by geohash
+    # so that polygons close to each other are numbered similarly.
+    polygons_with_unique_ids_sorted = polygons_with_unique_ids.sort_values(by=['UID']).reset_index()
+    polygons_with_unique_ids_sorted['WB_ID'] = polygons_with_unique_ids_sorted.index
+
+    # The step above creates an 'index' column, which we don't actually want, so drop it.
+    polygons_with_unique_ids_sorted.drop(columns=['index'], inplace=True)
+
+    # Reproject to the same crs as the input polygons.
+    polygons_with_unique_ids_sorted = polygons_with_unique_ids_sorted.to_crs(crs)
+
+    return polygons_with_unique_ids_sorted
+
+
+def get_timeseries_s3_uri(
+        uid,
+        product_version,
+        output_bucket_name,
+        ):
+    """
+    Get the timeseries s3 URI given a unique identifier for a polygon.
+
+    Parameters
+    ----------
+    uid : str
+        Unique identifier
+    product_version : str
+        The product version for the DE Africa Waterbodies service. 
+    output_bucket_name : str
+        The s3 bucket for the DE Africa Waterbodies service shapefiles and timeseries.
+
+    Returns
+    -------
+    str
+        A s3 URI for the timeseries for a waterbody polygon.
+    """
+
+    # Incase storage location is local.
+    if output_bucket_name is None:
+        output_bucket_name == "deafrica-waterbodies-dev"
+
+    version = product_version.replace(".", "-")
+
+    subfolder = uid[:4]
+
+    csv_file = f"{uid}_v{version[0]}.csv"
+
+    timeseries_s3_uri = f"s3://{output_bucket_name}/{version}/timeseries/{subfolder}/{csv_file}"
```

Comment:
The timeseries attribute needs to be a URL, rather than the s3 URI. This will then allow people to view it in DE Africa maps and download it. To me, it looks like it should have the form:

```
https://{output_bucket_name}.s3.af-south-1.amazonaws.com/{version}/timeseries/{subfolder}/{csv_file}
```

Reply:
I've also moved the waterbodies commands into a single script https://github.com/digitalearthafrica/deafrica-waterbodies/pull/3/commits/db51136644a14a5c058bf2dcdbc343d832b3b565